core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1574,36 +1616,52 @@ public void run() {
 
         @Override
         protected ManagedChannelBuilder<?> delegate() {
-          return managedChannelImplBuilder;
+          return delegate;
         }
 
         @Override
         public ManagedChannel build() {
           // TODO(creamsoup) prevent main channel to shutdown if oob channel is not terminated
-          return new ManagedChannelImpl(
-                  managedChannelImplBuilder,
-                  originalTransportFactory,
-                  backoffPolicyProvider,
-                  balancerRpcExecutorPool,
-                  stopwatchSupplier,
-                  Collections.<ClientInterceptor>emptyList(),
-                  timeProvider);
+          // TODO(zdapeng) register the channel as a subchannel of the parent channel in channelz.
+          if (managedChannelImplBuilder != null) {
+            return new ManagedChannelImpl(
+                managedChannelImplBuilder,
+                transportFactory,
+                backoffPolicyProvider,
+                balancerRpcExecutorPool,
+                stopwatchSupplier,
+                Collections.<ClientInterceptor>emptyList(),
+                timeProvider);
+          } else {
+            return delegate.build();
+          }
         }
       }
 
       checkState(!terminated, "Channel is terminated");
 
       @SuppressWarnings("deprecation")
-      ResolvingOobChannelBuilder builder = new ResolvingOobChannelBuilder(target)
+      ResolvingOobChannelBuilder builder = new ResolvingOobChannelBuilder(target, creds)
           .nameResolverFactory(nameResolverFactory);
 
       return builder
-          .overrideAuthority(getAuthority())
+          .overrideAuthority(ManagedChannelImpl.this.authority())
           .maxTraceEvents(maxTraceEvents)
           .proxyDetector(nameResolverArgs.getProxyDetector())
           .userAgent(userAgent);
     }
 
+    @Override
+    public ChannelCredentials getUnsafeChannelCredentials() {
+      if (originalChannelCreds == null) {
+        return null;
+      }
+      if (originalCallCreds == null) {
+        return originalChannelCreds;
+      }
+      return CompositeChannelCredentials.create(originalChannelCreds, originalCallCreds);
```

Comment:
The original channel creds should include the call creds. Don't combine them together. You should be able to delete the `originalCallCreds` field.

Reply:
Right. Done.